hudi-hadoop-mr/src/test/java/org/apache/hudi/hadoop/realtime/TestHoodieMergeOnReadSnapshotReader.java

```diff
@@ -81,6 +85,8 @@ public void setUp() {
     hadoopConf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
     baseJobConf = new JobConf(hadoopConf);
     baseJobConf.set(HoodieRealtimeConfig.MAX_DFS_STREAM_BUFFER_SIZE_PROP, String.valueOf(1024 * 1024));
+    baseJobConf.set(serdeConstants.LIST_COLUMNS, COLUMNS);
```

Comment:
how come the test was flaky w/o this. I would expect it to fail cosistently right?
I mean, even when the original author wrote the tests, it would encountered this issue right ?

Reply:
Yes NPE was reproducible locally. As per the logic in `prepareHiveAvroSerializer`, these configs are not strictly necessary and the execption is caught and execution proceeds with a flag turned off. These configs are mainly required for schema evolution.

Reply:
sg